src/shared_modules/keystore/src/keyStore.cpp

```diff
@@ -0,0 +1,128 @@
+#include "keyStore.hpp"
+#include "evpHelper.hpp"
+#include "loggerHelper.h"
+#include "rocksDBWrapper.hpp"
+#include "rsaHelper.hpp"
+
+// Database constants, based on the keystore path.
+constexpr auto DATABASE_PATH {"queue/keystore"};
+
+// File constants, used in the version 1 of the keystore, currently in use only for the upgrade to version 2, that uses
+// AES 256 encryption, without the need of the private key file.
+constexpr auto PRIVATE_KEY_FILE {"etc/sslmanager.key"};
+
+// Keystore constants, KS_NAME is used in the logs as tag.
+// KS_VERSION is the current version of the keystore. Used to identify the version of the keystore in the database.
+// KS_VERSION_FIELD is the field used to store the version of the keystore in the database.
+constexpr auto KS_NAME {"keystore"};
+constexpr auto KS_VERSION {"2"};
+constexpr auto KS_VERSION_FIELD {"version"};
+
+static void upgrade(Utils::RocksDBWrapper& keystoreDB, const std::string& columnFamily)
+{
+    std::string versionValue;
+    std::string rawValue;
+
+    // If the version field does not exist, it means that the keystore has not been upgraded yet.
+    if (!keystoreDB.get(KS_VERSION_FIELD, versionValue, columnFamily))
+    {
+        try
+        {
+            // Check if the private key file exists
+            if (!std::filesystem::exists(PRIVATE_KEY_FILE))
+            {
+                throw std::runtime_error("Private key file not found.");
+            }
+
+            // Upgrade all keys
+            for (const auto& [key, value] : keystoreDB.begin(columnFamily))
+            {
+                std::string encryptedRSAValue;
+                std::vector<char> encryptedValue;
+
+                // Get the encrypted RSA value
+                if (keystoreDB.get(key, encryptedRSAValue, columnFamily))
+                {
+                    logInfo(KS_NAME, "Upgrading %s key pair.", key.c_str());
+                }
+
+                // Decrypt the RSA value
+                RSAHelper().rsaDecrypt(PRIVATE_KEY_FILE, encryptedRSAValue, rawValue);
+
+                // Encrypt the value with AES 256
+                EVPHelper().encryptAES256(rawValue, encryptedValue);
+
+                // Insert the key-value pair using AES encryption
+                keystoreDB.put(key, rocksdb::Slice(encryptedValue.data(), encryptedValue.size()), columnFamily);
+            }
+        }
+        catch (const std::exception& exception)
+        {
+            // If the upgrade fails, delete all keys and log the error.
+            keystoreDB.deleteAll(columnFamily);
+            logWarn(KS_NAME, "Keystore upgrade failed, re-run this tool with all keys. Error: %s", exception.what());
```

Comment:
This warning doesn't appear if we run the keystore binary. We may want to print this to the cerr

Reply:
This warning appears when the method "Get" from the keystone class is called, and the indexer connector use this get method.

Reply:
It is part of the upgrade process, I am concerned about the behavior from the user's perspective

1. Have data stored using the old keystore in the column "a"
2. Certs are wrong for some reason
2. Use the new keystore in the column "a"
3. All the other values will be removed and the user was not warned

![image](https://github.com/user-attachments/assets/83479bc6-b676-4909-af49-69bd77809c8d)

> [!NOTE]
> I know that once the certs are lost, there is no way to recover them but te user can't tell that this happen

Reply:
Your point makes sense, I'll add it.